### PR TITLE
remove traces of oA

### PR DESCRIPTION
**Pull request by jschmid1**
Deepsea does not support openATTIC is future releases. This patch removes all traces including all traces of apache/httpd-prefork but:

- remove.openattic
- rescind.openattic

from the codebase

In addition I adapted the openattic validation to raise an error when a role->openattic assignment is detected in the cluster. This will advise the user to run the remove/rescind command to remove openattic.

Signed-off-by: Joshua Schmid <jschmid@suse.de>

-----------------

**Checklist:**
- [x] Adapted documentation
- [x] Referenced issues or internal bugtracker
- [x] Ran integration tests successfully (trigger with "@susebot run teuthology" in a GitHub comment; see the [wiki](https://github.com/SUSE/DeepSea/wiki/Testing) for more information)

### CODE REVIEW

**Comment by jschmid1** [comment]
@susebot run teuthology

**Comment by smithfarm** [comment]
Good to see this!

File: deepsea.spec.in [hunk 1]
```diff
@@ -644,10 +620,10 @@ systemctl try-restart salt-api > /dev/null 2>&1 || :
 %config /srv/salt/ceph/rescind/time/chrony/*.sls
 %config /srv/salt/ceph/rescind/time/ntp/*.sls
 %config /srv/salt/ceph/rescind/tuned/*.sls
-%config /srv/salt/ceph/rescind/openattic/*.sls
+%config /srv/salt/ceph/rescind/openattic/*.sls
```
Comment (tserong) [inline deepsea.spec.in]: Minor nit: trailing whitespace

File: srv/salt/_modules/cephprocesses.py [hunk 2]
```diff
@@ -223,9 +223,7 @@ def add(self, prc, role):
         Also do a special check for oA
```
Comment (tserong) [inline srv/salt/_modules/cephprocesses.py]: Can the above mention of oA also go away?

File: srv/salt/_modules/fs.py [hunk 3]
```diff
@@ -109,7 +109,7 @@ def _ceph_is_down():
     retries = 6
     delay = 2
     down = False
-    # Processes that don't impede migration (httpd-prefork == openattic, rest are ganesha related).
+    # Processes that don't impede migration (ganesha related).
     omit_list = ['httpd-prefork', 'ganesha.nfsd', 'rpcbind', 'rpc.statd']
```
Comment (tserong) [inline srv/salt/_modules/fs.py]: Can we remove httpd-prefork from this list?

File: srv/salt/ceph/remove/openattic/default.sls [hunk 4]
```diff
@@ -9,8 +9,3 @@ remove openattic auth:
     - name: "ceph auth del client.openattic"
 
 {% endif %}
-
-fix salt job cache permissions:
-  cmd.run:
-  - name: "find /var/cache/salt/master/jobs -user root -exec chown {{ salt['deepsea.user']() }}:{{ salt['deepsea.group']() }} {} ';'"
-
```
Comment (tserong) [inline srv/salt/ceph/remove/openattic/default.sls]: Why drop this?

**Comment by LenzGr** [comment]
> ui_iscsi.py is gone, but ui_rgw.py and ui_ganesha.py remain. Should the latter two be removed as well?

I suggest to ping @ricardoasmarques and @rjfd about that - the Ceph Manager dashboard may still use this.

**Comment by rjfd** [comment]
> ui_iscsi.py is gone, but ui_rgw.py and ui_ganesha.py remain. Should the latter two be removed as well?

I think we can remove all runners used exclusively by openATTIC, and if the ceph-dashboard ends up requiring them, then we will re-add the code in another PR.

File: srv/salt/_modules/fs.py (hunk 3 above)
Comment (jschmid1) [reply srv/salt/_modules/fs.py]: right, httpd-prefork should be excluded now.. Let me check that again though

File: srv/salt/ceph/remove/openattic/default.sls (hunk 4 above)
Comment (jschmid1) [reply srv/salt/ceph/remove/openattic/default.sls]: I'm fairly certain that salt2018 has fixed this issue internally. We can remove all other occurrences if that statement asserts true.

File: srv/salt/_modules/cephprocesses.py (hunk 2 above)
Comment (jschmid1) [reply srv/salt/_modules/cephprocesses.py]: right, I grepped for openattic and omitted oA :/

File: deepsea.spec.in (hunk 1 above)
Comment (jschmid1) [reply deepsea.spec.in]: fixing

**Comment by jschmid1** [comment]
> I think we can remove all runners used exclusively by openATTIC, and if the ceph-dashboard ends up requiring them, then we will re-add the code in another PR.

even better! I'll remove those files aswell

**Comment by smithfarm** [comment]
@susebot run teuthology deepsea/tier3/functional

File: srv/salt/_modules/cephprocesses.py (hunk 2 above)
Comment (tserong) [reply srv/salt/_modules/cephprocesses.py]: :-)

File: srv/salt/ceph/remove/openattic/default.sls (hunk 4 above)
Comment (tserong) [reply srv/salt/ceph/remove/openattic/default.sls]: Oh, cool. If that's fixed internally to salt I'll be very happy.

**Comment by smithfarm** [comment]
@susebot run teuthology deepsea/tier3/functional

**Comment by jschmid1** [comment]
@susebot run teuthology deepsea/tier3/functional

**Comment by jschmid1** [comment]
removed `ui_rgw`, `ui_ganesha` aswell

**Pull request merged**